modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp

```diff
@@ -292,13 +296,48 @@ bool GenericConstraintSolver::buildSystem(const core::ConstraintParams *cParams,
         sofa::helper::AdvancedTimer::stepBegin("Get Compliance");
         msg_info() <<" computeCompliance in "  << constraintCorrections.size()<< " constraintCorrections" ;
 
-        for (unsigned int i=0; i<constraintCorrections.size(); i++)
-        {
-            core::behavior::BaseConstraintCorrection* cc = constraintCorrections[i];
-            if (!cc->isActive()) continue;
-            sofa::helper::AdvancedTimer::stepBegin("Object name: " + cc->getName());
-            cc->addComplianceInConstraintSpace(cParams, &current_cp->W);
-            sofa::helper::AdvancedTimer::stepEnd("Object name: " + cc->getName());
+        if(multithreading.getValue()){
+            helper::vector<std::thread> threads(constraintCorrections.size());
+            helper::vector<sofa::component::linearsolver::LPtrFullMatrix<double>> Ws;
+            Ws.resize(constraintCorrections.size());
+            helper::vector<core::ConstraintParams> Ps;
+            Ps.resize(constraintCorrections.size());
+
+            for (unsigned int i=0; i<constraintCorrections.size(); i++)
+            {
+                Ws[i].resize(current_cp->W.rowSize(),current_cp->W.colSize());
+                Ps[i] = *cParams;
+                auto Wi = &Ws[i];
+                auto Pi = Ps[i];
+                core::behavior::BaseConstraintCorrection* cc = constraintCorrections[i];
+                threads[i] = std::thread([Pi, Wi, cc] {
+                    if (cc->isActive()) {
+                        cc->addComplianceInConstraintSpace(&Pi, Wi);
+                    }
+                });
+            }
+            for (auto& t : threads)
+                t.join();
+
+            for (unsigned int i=0; i<constraintCorrections.size(); i++)
```

Comment:
I'm not sure if it will change something here because of the virtual calls to `current_cp->W.add()` and `Ws[i].element()`, but in time sensitive loops, I usually tend to strip out functions calls from the end condition of the loop. This will allow the compiler to be certain that the end condition will not change during the loop. i.e.:

```cpp
auto & W = current_cp->W;
const auto rows = W.rowSize();
const auto cols = W.colSize();
const auto n_constraintCorrections = constraintCorrections.size();

// Accumulate the contribution of each constraint corrections 
// into the system's compliant matrix W
for (sofa::Index i = 0; i < n_constraintCorrections; i++) {
    const auto * cc = constraintCorrections[i];
    const auto & Wi = Ws[i];
    
    if (!cc->isActive()) {
        continue;
    }
    
    for (sofa::Index j = 0; j < rows; ++j) {
        for (sofa::Index l = 0; l < cols; ++l) {
            W.add(j, l, Wi.element(j,l));
        }
    }
}
```

I also refactored a little bit the code to the format I usually like to read, feel free to take/reject whatever suits you ;)